src/app/shared/components/button/button.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
+import { IconName, IconProp } from '@fortawesome/fontawesome-svg-core';
 import { fas } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.scss']
 })
-export class ButtonComponent {
+export class ButtonComponent{
+  @Input() buttonText?:string;
+  @Input() iconName?:'trash-can' | 'pencil';
+  icon: IconProp | undefined;
+  
   constructor(library: FaIconLibrary) {
     library.addIconPacks(fas);
```

Comment:
Am I right to think that you are importing the full package (with why know how many svg-s) not just the trash can and pencil? Can we make sure to import only those we need?

Reply:
Yes , you are right, but this is the way how we got it, how the forked repo implements it this is why i did not change.
Should I change general library usage to the 2 specific imports?

Reply:
Dont waste time on it, just note that we can restrict the import to individual icons. Probably I would not be so excusing on a production grade project 😄